Assets/MixedRealityToolkit/UX/Scripts/BoundingBoxes/BoundingBoxRig.cs

```diff
@@ -0,0 +1,470 @@
+            appBarInstance.BoundingBox = boxInstance;
+
+            boxInstance.IsVisible = false;
+        }
+        private void Update()
+        {
+            if (destroying == false && ShowRig)
+            {
+                UpdateBoundsPoints();
+                UpdateHandles();
+            }
+        }
+
+        private void OnDestroy()
+        {
+            destroying = true;
+            ShowRig = false;
+            ClearHandles();
+        }
+
+        private void UpdateBoundsPoints()
+        {
+            handleCentroids = GetBounds();
+        }
+
+        private void CreateHandles()
+        {
+            ClearHandles();
+            UpdateCornerHandles();
+            UpdateRotateHandles();
+            ParentHandles();
+            UpdateHandles();
+        }
+
+        private void UpdateCornerHandles()
+        {
+            handleCentroids = handleCentroids ?? GetBounds();
+
+            if (cornerHandles == null)
+            {
+                cornerHandles = new GameObject[handleCentroids.Count];
+                for (int i = 0; i < cornerHandles.Length; ++i)
+                {
+                    cornerHandles[i] = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                    cornerHandles[i].GetComponent<Renderer>().material = scaleHandleMaterial;
+                    cornerHandles[i].transform.localScale = scaleHandleSize;
+                    cornerHandles[i].AddComponent<BoundingBoxGizmoHandle>();
+                    cornerHandles[i].GetComponent<BoundingBoxGizmoHandle>().Rig = this;
+                    cornerHandles[i].GetComponent<BoundingBoxGizmoHandle>().TransformToAffect = objectToBound.transform;
+                    cornerHandles[i].GetComponent<BoundingBoxGizmoHandle>().Axis = BoundingBoxGizmoHandle.AxisToAffect.Y;
+                    cornerHandles[i].GetComponent<BoundingBoxGizmoHandle>().AffineType = BoundingBoxGizmoHandle.TransformType.Scale;
+                    cornerHandles[i].name = "Corner " + i.ToString();
+                }
+            }
+
+            for (int i = 0; i < cornerHandles.Length; ++i)
+            {
+                cornerHandles[i].transform.position = handleCentroids[i];
+                cornerHandles[i].transform.localRotation = objectToBound.transform.rotation;
+            }
+        }
+
+        private void UpdateRotateHandles()
+        {
+            handleCentroids = handleCentroids ?? GetBounds();
+
+            if (rotateHandles == null)
+            {
+                rotateHandles = new GameObject[12];
+
+                for (int i = 0; i < rotateHandles.Length; ++i)
+                {
+                    rotateHandles[i] = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+                    rotateHandles[i].GetComponent<Renderer>().material = rotateHandleMaterial;
+                    rotateHandles[i].transform.localScale = rotateHandleSize;
+                    rotateHandles[i].AddComponent<BoundingBoxGizmoHandle>();
```

Comment:
cache and use the handle component.

Reply:
You could even make a cache for each handle and re-use it outside the for loop below so you don't have to use get component on each again.